test/e2e/multihoming.go

```diff
@@ -91,35 +92,38 @@ var _ = Describe("Multi Homing", func() {
 				}
 				return updatedPod.Status.Phase
 			}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
+			podIPForAttachment(cs, pod.GetNamespace(), pod.GetName(), secondaryNetworkName, 0)
 
-			if netConfig.excludeCIDRs != nil {
-				podIP, err := podIPForAttachment(cs, pod.GetNamespace(), pod.GetName(), secondaryNetworkName, 0)
-				Expect(err).NotTo(HaveOccurred())
-				subnet, err := getNetCIDRSubnet(netConfig.cidr)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(inRange(subnet, podIP)).To(Succeed())
-				for _, excludedRange := range netConfig.excludeCIDRs {
-					Expect(inRange(excludedRange, podIP)).To(
-						MatchError(fmt.Errorf("ip [%s] is NOT in range %s", podIP, excludedRange)))
+			if len(netConfig.excludeCIDRs) > 0 {
+				for i := range len(strings.Split(netConfig.cidr, ",")) {
+					podIP, err := podIPForAttachment(cs, pod.GetNamespace(), pod.GetName(), secondaryNetworkName, i)
+					Expect(err).NotTo(HaveOccurred())
+					foundRange := false
+					for _, cidr := range strings.Split(netConfig.cidr, ",") {
+						subnet, err := getNetCIDRSubnet(cidr)
+						Expect(err).NotTo(HaveOccurred())
+						if err := inRange(subnet, podIP); err == nil {
+							// ip was found in one of the network ranges
+							foundRange = true
+							break
+						}
+					}
+					Expect(foundRange).To(BeTrue())
+					for _, excludedRange := range strings.Split(netConfig.excludeCIDRs, ",") {
+						Expect(inRange(excludedRange, podIP)).To(
+							MatchError(fmt.Errorf("ip [%s] is NOT in range %s", podIP, excludedRange)))
+					}
```

Comment:
Asking to undersntad, this code is now necessary because `cidr` could end up having multiple CIDRs?

I think it would make the test easier to follow if this code or some part of it to function.

Reply:
This is no longer needed, I was unaware that secondary networks can have any IP family regardless of the underlying cluster